### PR TITLE
[#179659371] Return nonce of default resolver when domain doesn't have own

### PR DESCRIPTION
**Pull request by hivtushok**

### CODE REVIEW

File: test/metatx/ResolverForwarder.test.js [hunk 1]
```diff
@@ -55,6 +60,24 @@ describe('ResolverForwarder', () => {
...
       expect(nonceF).to.be.equal(nonceR);
     });
+
+    it(`should return nonce of default resolver when domain doesn't have own`, async () => {
+      const tokenId = await mintDomainWithoutResolver('test_foo_no_res', owner.address);
+      let nonceF = await forwarder.nonceOf(tokenId);
+      let nonceR = await resolver.nonceOf(tokenId);
+
+      expect(nonceF).to.be.equal(0);
+      expect(nonceF).to.be.equal(nonceR);
```
Comment (bogdan) [inline test/metatx/ResolverForwarder.test.js:70]:
```suggestion
      expect(nonceF).to.be.equal(nonceR);
      expect(await registry.resolverOf(tokenId)).to.be.equal(
        "0x0000000000000000000000000000000000000000"
      )
```

**Review by bogdan — APPROVED** [review]

File: test/metatx/ResolverForwarder.test.js (hunk 1 above)
Comment (hivtushok) [reply test/metatx/ResolverForwarder.test.js]: Resolved with 'to-be-rejected' since this call reverts the transaction for domain without resolver.

**Review by bogdan — APPROVED** [review]

File: contracts/metatx/ResolverForwarder.sol [hunk 2]
```diff
@@ -14,17 +14,25 @@ import './BaseRoutingForwarder.sol';
  */
 contract ResolverForwarder is BaseRoutingForwarder {
     ICNSRegistry private _cnsRegistry;
+    address private _defaultCnsResolverAddress;
```
Comment (aquiladev) [inline contracts/metatx/ResolverForwarder.sol:17]: I would suggest to use shorter name  `_defaultResolver`, `_defaultCnsResolver`

File: contracts/metatx/ResolverForwarder.sol [hunk 3]
```diff
@@ -14,17 +14,25 @@ import './BaseRoutingForwarder.sol';
...
 
-    constructor(ICNSRegistry cnsRegistry) {
+    constructor(ICNSRegistry cnsRegistry, address defaultCnsResolverAddress) {
         _cnsRegistry = cnsRegistry;
+        _defaultCnsResolverAddress = defaultCnsResolverAddress;
```
Comment (aquiladev) [inline contracts/metatx/ResolverForwarder.sol:21]: same here, short name is better

File: contracts/metatx/ResolverForwarder.sol [hunk 4]
```diff
@@ -14,17 +14,25 @@ import './BaseRoutingForwarder.sol';
...
     }
 
     function nonceOf(uint256 tokenId) public view override returns (uint256) {
-        IForwarder target = IForwarder(_cnsRegistry.resolverOf(tokenId));
+        address addr;
+         try _cnsRegistry.resolverOf(tokenId) returns (address _resolver) {
+            addr = _resolver;
+        } catch {
+            addr = _defaultCnsResolverAddress;
+        }
```
Comment (aquiladev) [inline contracts/metatx/ResolverForwarder.sol:34]:
```suggestion
        address resolver = _defaultCnsResolverAddress;
         try _cnsRegistry.resolverOf(tokenId) returns (address _resolver) {
            resolver = _resolver;
        } catch { }
```

File: test/metatx/ResolverForwarder.test.js [hunk 5]
```diff
@@ -55,6 +60,16 @@ describe('ResolverForwarder', () => {
       expect(nonceF).to.be.equal(1);
       expect(nonceF).to.be.equal(nonceR);
     });
+
+    it(`should return nonce of default resolver when domain doesn't have own`, async () => {
```
Comment (aquiladev) [inline test/metatx/ResolverForwarder.test.js:64]:
```suggestion
    it(`should return nonce from default resolver when domain doesn't have own`, async () => {
```

**Review by aquiladev — APPROVED** [review]

**Pull request merged**